### DISCUSSION TITLE
动态where条件

### DISCUSSION THREAD

**Discussion by liusn0329** [wgzhao/Addax#170] [Q&A]
背景：通过一个脚本抽取多个库中的表，json中的tableName、jdbcUrl、where条件都是动态传入的。
python datax.py  test.json -p -DtableName=xxx  -DjdbcUrl=xxxx  -Dwhere=testDate>'2021-4-20' and testDate<'2021-5-20'
问题：日志打出来的where条件为testDate>'2021-4-20' and testDate<'2021-5-20，丢失了最右侧的单引号，导致sql报错。
请问这个应该怎么解决？

**Comment by liusn0329** [comment]
@wgzhao

**Reply by wgzhao** [reply]
```shell
python datax.py \
 -p "-DtableName=xxx -DjdbcUrl=xxxx -Dwhere=testDate>'2021-4-20' and testDate<'2021-5-20'" \
 test.json
```

这样试试

**Reply by liusn0329** [reply]
问题找到了，不是python脚本的问题，是通过commons-exec去执行脚本的时候，CommandLine.parse()把末尾的单引号去掉了
